src/libcore/macros.rs

```diff
@@ -302,6 +302,7 @@ macro_rules! debug_assert_ne {
 /// ```
 #[macro_export]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_deprecated(since = "1.39.0", reason = "use the `?` operator instead")]
```

Comment:
Notice that this means that nightly will start warning in a week -- which is an entire cycle *before* the deprecation is announced on the release notes.

We usually deprecate 3 releases in the future, so that between the deprecation announcement on a stable release and nightly starting to warn, there is still a full cycle. (Well, "usually" as in, both times I was involved in a future deprecation this is what we did -- in one case after some pushback when we wanted to go faster.)

Reply:
I don't think that matters much; `try!` has been deprecated a long time in the docs and since `?` was stabilized -- the change here is only due to a bugfix in the language.